### PR TITLE
Suppress pipeline construction error message for PEFT model

### PR DESCRIPTION
**Pull request by B-Step62**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/B-Step62/mlflow/pull/11187?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11187/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11187
```

</p>
</details>

### What changes are proposed in this pull request?

Transformers pipeline emits the following error message when a pipeline is constructed with a PEFT model.

> The model 'PeftModelForCausalLM' is not supported for text-generation. Supported models are ['BartForCausalLM', 'BertLMHeadModel', ...

This is (mostly) false alert because PEFT model works equally as the wrapped model that is supported by the pipeline. This message is emitted as error log ([source](https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/base.py#L1042-L1045)) but it doesn't block us to create pipeline. We can simply suppress that error to avoid confusion. 

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/logging_utils.py [hunk 1]
```diff
@@ -87,3 +89,31 @@ def _configure_mlflow_loggers(root_module_name):
...
+        self._module = module
+
+    def filter(self, record):
+        if record.name == self._module and self._pattern.search(record.msg):
+            return False
+        return True
+
+
+@contextlib.contextmanager
+def suppress_logs(module: str, filter_regex: str):
```
Comment (harupy) [inline mlflow/utils/logging_utils.py:107]:
```suggestion
def suppress_logs(module: str, filter_regex: re.Pattern):
```

can we pre-compile the regex?

File: mlflow/transformers/__init__.py [hunk 2]
```diff
@@ -121,6 +122,10 @@
 if TYPE_CHECKING:
     from transformers import Pipeline
 
+# Transformers pipeline complains that PeftModel is not supported for any task type, even
+# when the wrapped model is supported. As MLflow require users to use pipeline for logging,
+# we should suppress that confusing error message.
+_PEFT_PIPELINE_ERROR_MSG = r"The model '(PeftModel[^']*)' is not supported for ([^.]+)\."
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/transformers/__init__.py:128]: Do we need `[^.]+\.`?

[the remark this answers is not in the document]
Comment (B-Step62) [reply mlflow/transformers/__init__.py:128]: Not really, can simplify the regex

**Review by harupy — APPROVED** [review]
LGTM!

**Review by BenWilson2 — APPROVED** [review]
LGTM!

**Pull request merged**